core/src/impl/Kokkos_HostSpace.cpp

```diff
@@ -119,30 +87,13 @@ void *HostSpace::impl_allocate(
   void *ptr = nullptr;
 
   if (arg_alloc_size) {
-    if (m_alloc_mech == STD_MALLOC) {
-      // Over-allocate to and round up to guarantee proper alignment.
-      size_t size_padded = arg_alloc_size + sizeof(void *) + alignment;
-
-      void *alloc_ptr = malloc(size_padded);
-
-      if (alloc_ptr) {
-        auto address = reinterpret_cast<uintptr_t>(alloc_ptr);
-
-        // offset enough to record the alloc_ptr
-        address += sizeof(void *);
-        uintptr_t rem    = address % alignment;
-        uintptr_t offset = rem ? (alignment - rem) : 0u;
-        address += offset;
-        ptr = reinterpret_cast<void *>(address);
-        // record the alloc'd pointer
-        address -= sizeof(void *);
-        *reinterpret_cast<void **>(address) = alloc_ptr;
-      }
-    }
-#if defined(KOKKOS_ENABLE_INTEL_MM_ALLOC)
-    else if (m_alloc_mech == INTEL_MM_ALLOC) {
-      ptr = _mm_malloc(arg_alloc_size, alignment);
-    }
+    // Over-allocate to to multiple of alignment
+    size_t size_alignment_multiple =
+        (arg_alloc_size + alignment - 1) / alignment * alignment;
+#ifdef KOKKOS_COMPILER_MSVC
+    ptr = _aligned_malloc(size_alignment_multiple, alignment);
```

Comment:
Does `_aligned_malloc` also require the size argument to be a multiple of the alignment?

Reply:
I couldn't find information about that but preferred doing it anyway for consistency.

Reply:
Does MSVC not support aligned_alloc?

Reply:
MSVC indeed does not support `aligned_alloc` as noted [on CppReference.com](https://en.cppreference.com/w/cpp/memory/c/aligned_alloc)

It would be nice to be able to use `operator new(size_alignment_multiple, std::align_val_t(alignment))` instead, avoiding the conditional, but then we'd have to store the alignment to pass to the corresponding `operator delete`